Add setup instructions for Kong Gateway (proxy) (#18291)
DataDog/documentation: content/en/tracing/trace_collection/proxy_setup/_index.md

[12]: https://istio.io/docs/ops/configuration/traffic-management/protocol-selection/#manual-protocol-selection
[13]: https://istio.io/latest/docs/releases/supported-releases/#support-status-of-istio-releases
{{% /tab %}}

> {{% /tab %}}

Reviewer: ```suggestion
```

{{% tab "Kong" %}}

Datadog APM is available for [Kong Gateway][1] using the [kong-plugin-ddtrace][2] plugin.

## Installation

The plugin is installed using `luarocks`.
```
luarocks install kong-plugin-ddtrace
```

> As this is not a bundled plugin, Kong Gateeway needs to be configured to enable it.
> This can be done by including `bundled` and `ddtrace` in the `KONG_PLUGINS` environment variable, or by setting `plugins=bundled,ddtrace` in `/etc/kong/kong.conf`
> Kong Gateway should be restarted after this change is made.

Reviewer: ```suggestion
Kong Gateway is not a bundled plugin, so it needs to be configured before it can be enabled.
To enable it, include `bundled` and `ddtrace` in the `KONG_PLUGINS` environment variable, or set `plugins=bundled,ddtrace` in `/etc/kong/kong.conf`. Next, restart Kong Gateway to apply the change.
```

```
# Set the KONG_PLUGINS environment variable or edit /etc/kong/kong.conf to enable the ddtrace plugin
export KONG_PLUGINS=bundled,ddtrace
kong restart
```

## Configuration

The plugin can be enabled globally or on specific services in Kong Gateway.

```
# Enabled globally
curl -i -X POST --url http://localhost:8001/plugins/ --data 'name=ddtrace'
# Enabled for specific service only
curl -i -X POST --url http://localhost:8001/services/example-service/plugins/ --data 'name=ddtrace'
```

Options are available for setting the service name, environment, and other features within the plugin.
The example below sets the service name to `mycorp-internal-api` in the `prod` environment.
```
curl -i -X POST --url http://localhost:8001/plugins/ --data 'name=ddtrace' --data 'config.service_name=mycorp-internal-api' --data 'config.environment=prod'
```

More configuration options can be found on the [kong-plugin-ddtrace][3] plugin documentation.

[1]: https://docs.konghq.com/gateway/latest/
[2]: https://github.com/DataDog/kong-plugin-ddtrace
[3]: https://github.com/DataDog/kong-plugin-ddtrace#configuration

> 

Reviewer: ```suggestion
{{% /tab %}}
```
